lib/movements.js

```diff
@@ -103,6 +105,48 @@ class Movements {
     return null
   }
 
+  // Get any entities who's bounding box intersects the node + offset
+  // Could work impropperly if the entity happens to walk exactly along the path and gets detected again but, this probably won't happen. Maybe in the future there could be a sorted list of ents to select from and also a list of detected ents
+  getIntersectingEntities (pos, dx, dy, dz) {
+    const ents = []
+    if (this.allowEntityDetection && pos) {
+      const MAX_DIST_SQ = 36 // Needs to be this high b/c we pretend the floored coords of the block are a 2x2 space, then we take into account the max 'typical' bounding box size of 4x4
+      const dPos = new Vec3(pos.x + dx, pos.y + dy, pos.z + dz).floored()
+      for (const ent of Object.values(this.bot.entities)) {
+        if (ent !== this.bot.entity && ent.type !== 'item') {
+          const entPos = ent.position.offset(ent.velocity.x, ent.onGround * ent.velocity.y, ent.velocity.z)
+          if (entPos.distanceSquared(dPos) <= MAX_DIST_SQ) {
+            const maxYCoord = entPos.y + ent.height
+            if ((entPos.y <= dPos.y && maxYCoord > dPos.y) || (entPos.y >= dPos.y && entPos.y < dPos.y + 1.0)) { // Check if ent position below block but bounding box extends into block or the entity's feet are in the block
+              const halfWidth = ent.width / 2.0
+              const entBoundsPos = entPos.plus(new Vec3(halfWidth, 0, halfWidth))
+              const entBoundsNeg = entPos.minus(new Vec3(halfWidth, 0, halfWidth))
+              if (((entPos.x >= dPos.x && entPos.x < dPos.x + 1.0) && (entPos.z >= dPos.z && entPos.z < dPos.z + 1.0)) || (((entBoundsPos.x >= dPos.x && entBoundsPos.x < dPos.x + 1.0) || (entBoundsNeg.x >= dPos.x && entBoundsNeg.x < dPos.x + 1.0)) && ((entBoundsPos.z >= dPos.z && entBoundsPos.z < dPos.z + 1.0) || (entBoundsNeg.z >= dPos.z && entBoundsNeg.z < dPos.z + 1.0)))) {
+                ents.push(ent)
+              }
+            }
+          }
+        }
+      }
+    }
+    return ents
+  }
+
+  // Faster but doesn't work good for block placement checks. Better for the entity cost system where we want to see if the actual entity is there
+  getEntitiesAt (pos, dx, dy, dz) {
+    const ents = []
+    if (this.allowEntityDetection && pos) {
+      const dPos = new Vec3(pos.x + dx, pos.y + dy, pos.z + dz).floored()
+      for (const ent of Object.values(this.bot.entities)) {
+        const entPos = ent.position.offset(ent.velocity.x, ent.onGround * ent.velocity.y, ent.velocity.z)
```

Comment:
`ent.onGround * ent.velocity.y` is this a typo? onGround should be a boolean right? Even it this might work you should use something like `ent.onGround ? 0 : ent.velocity.y` instead for better readability.

Reply:
Also why is this even taking entity velocity into account? Why not just check for the entity position and update the path when the bot is about to intersect the entity?

Reply:
> Also why is this even taking entity velocity into account? Why not just check for the entity position and update the path when the bot is about to intersect the entity?

So the reason I did this was because I thought that by the time the bot has finished running everything, the entities will probably already be in their next position. So instead of error correcting in a future tick it should just estimate their next position ahead of time since it shouldn't cost much more to do so.